suitcase/tiff/__init__.py

```diff
@@ -1,25 +1,256 @@
+        self._file_prefix = file_prefix
+        self._templated_file_prefix = ''
+        self._kwargs = kwargs
+        self._start_found = False
+
+    def start(self, doc):
+        '''Extracts `start` document information for formatting file_prefix.
+
+        This method checks that only one `start` document is seen and formats
+        `file_prefix` based on the contents of the `start` document.
+
+        Parameters:
+        -----------
+        doc : dict
+            RunStart document
+        '''
+
+        # raise an error if this is the second `start` document seen.
+        if self._start_found:
+            raise RuntimeError(
+                "The serializer in suitcase.tiff expects documents from one "
+                "run only. Two `start` documents where sent to it")
+        else:
+            self._start_found = True
+
+        # format self._file_prefix
+        self._templated_file_prefix = self._file_prefix.format(**doc)
+
+        # return the start document
+        return doc
+
+    def descriptor(self, doc):
+        '''Use `descriptor` doc to map stream_names to descriptor uid's.
+
+        This method usess the descriptor document information to map the
+        stream_names to descriptor uid's.
+
+        Parameters:
+        -----------
+        doc : dict
+            EventDescriptor document
+        '''
+        # extract some useful info from the doc
+        stream_name = doc.get('name')
+        self._stream_names[doc['uid']] = stream_name
+
+        # return the descriptor doc
+        return doc
+
+    def event_page(self, doc):
+        '''Add event page document information to a ".tiff" file.
+
+        This method adds event_page document information to a ".tiff" file,
+        creating it if nesecary.
+
+        .. warning::
+
+            All non 2D 'image like' data is explicitly ignored.
+
+        .. note::
+
+            The data in Events might be structured as an Event, an EventPage,
+            or a "bulk event" (deprecated). The DocumentRouter base class takes
+            care of first transforming the other repsentations into an
+            EventPage and then routing them through here, so no further action
+            is required in this class. We can assume we will always receive an
+            EventPage.
+
+        Parameters:
+        -----------
+        doc : dict
+            EventPage document
+        '''
+        event_model.verify_filled(doc)
+        stream_name = self._stream_names[doc['descriptor']]
+        for field in doc['data']:
+            for img in doc['data'][field]:
+                # check that the data is 2D, if not ignore it
+                if numpy.asarray(img).ndim == 2:
+                    # create a file for this stream and field if none exists
+                    if not self._files.get(stream_name, {}).get(field):
+                        filename = f'{self._templated_file_prefix}' +\
```

Comment:
Prefer parents to backslash for line continuation, [per PEP8](https://www.python.org/dev/peps/pep-0008/#maximum-line-length). Also, no need for `+` when concatenating two strings across lines. They will just implicitly concatenate.

Reply:
actually when using f-strings this is not true. you need to start each line with `f'..'` and the string concatenation across lines then doesn't work. I have implemented this in a different way which you hopefully find more pleasing !-).

Reply:
It works for me. Not sure what you mean. Example:

```py
In [1]: (f'asdf'
   ...:  f'asdf')
   ...:  
Out[1]: 'asdfasdf'
```

Reply:
ah, you put the brackets around it I was not. That resolves the issue I was having

Reply:
done